diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/CollapseIfStatementsRule.kt

```diff
@@ -0,0 +1,159 @@
+
+        var currNode = findNestedIf(node) ?: return
+        var nestedLevel = 2
+        var nestedIfNode :ASTNode?
+        while (true) {
+            nestedIfNode = findNestedIf(currNode)
+            if (nestedIfNode != null) {
+                nestedLevel++
+                currNode = nestedIfNode
+            } else {
+                break
+            }
+        }
+        if (nestedLevel < startCollapseFromLevel) {
+            return
+        }
+        // Since the external `if` statement is not the direct parent,
+        // we need multiple steps to take the required one
+        // BLOCK -> THEN -> IF
+        var currParentNode = currNode.treeParent.treeParent.treeParent
+        // Now collapse in reverse order
+        while (nestedLevel != 1 && isFixMode) {
+            Warnings.COLLAPSE_IF_STATEMENTS.warnAndFix(
+                configRules, emitWarn, isFixMode,
+                "avoid using redundant nested if-statements", currNode.startOffset, currNode
+            ) {
+                collapse(currParentNode, currNode)
+                currNode = currParentNode
+                currParentNode = currParentNode.treeParent.treeParent.treeParent
+                nestedLevel--
+            }
+        }
+    }
+
+    private fun findNestedIf(parentNode: ASTNode) : ASTNode? {
+        val parentThenNode = (parentNode.psi as KtIfExpression).then?.node
+        val nestedIfNode = parentThenNode?.findChildByType(IF) ?: return null
+        // We won't collapse statements, if nested `if` statement have `else` node
+        (nestedIfNode.psi as KtIfExpression).`else`?.node?.let {
+            return null
+        }
+        // We monitor which types of nodes are followed before nested `if`
+        // and we allow only a limited number of types to pass through.
+        // Otherwise discovered `if` it is not nested
+        val listOfNodesBeforeNestedIf = parentThenNode.getChildren(null).takeWhile { it.elementType != IF }
+        val allowedTypes = listOf(LBRACE, WHITE_SPACE, RBRACE, KDOC, BLOCK_COMMENT, EOL_COMMENT)
+        if (listOfNodesBeforeNestedIf.any { it.elementType !in allowedTypes }) {
+            return null
+        }
+        return nestedIfNode
+    }
+
+    private fun collapse(parentNode : ASTNode, nestedNode : ASTNode) {
+        collapseConditions(parentNode, nestedNode)
+        collapseThenBlocks(parentNode, nestedNode)
+    }
+
+    private fun collapseConditions(parentNode : ASTNode, nestedNode : ASTNode) {
+        // Merge parent and nested conditions
+        val parentCondition = (parentNode.psi as KtIfExpression).condition?.text
+        val nestedCondition = (nestedNode.psi as KtIfExpression).condition
+        lateinit var mergeCondition : String
+        // If the nested condition is compound,
+        // we need to put it to the brackets, according algebra of logic
+        if (nestedCondition?.node?.elementType == BINARY_EXPRESSION &&
```

Comment:
```suggestion
        // If the nested condition is compound,
        // we need to put it to the brackets, according algebra of logic
        val mergeCondition = if (nestedCondition?.node?.elementType == BINARY_EXPRESSION &&
```
You can use returned value of `if` expression to assign immutable ariable

Reply:
Done in [assign to if expr](https://github.com/cqfn/diKTat/pull/814/files#diff-251ce198ca409fe76ce1e2c0c723f67bf4f143f4351c163c1a86175f5be5f397R116)